src/spyglass/spikesorting/spikesorting_curation.py

```diff
@@ -533,13 +546,54 @@ def _get_peak_channel(waveform_extractor: si.WaveformExtractor, peak_sign: str,
     peak_channel = {key : int(val) for key, val in peak_channel_dict.items()}
     return peak_channel
 
+def _get_cosine_similarity(waveform_extractor: si.WaveformExtractor, **metric_params):
+    """Gets the cosine similarity as computed by SpikeInterface's compute_template_similarity
+    """
+    cos_sim_mat = si.postprocessing.compute_template_similarity(
```

Comment:
This function seems to compute cosine similarity between waveforms at all channels that have been concatenated. This is not ideal for devices with many channels (e.g. probes) in which only a handful of channels actually has the signal - in that case the cosine similarity will be dominated by noise. I would suggest writing our own function that restricts the channel range based on probe geometry (and turn that into a PR to spikeinterface). I can help if needed.

Reply:
https://github.com/SpikeInterface/spikeinterface/blob/4633750f609aa66d91b8d3c25f9dfc5b7ca06144/spikeinterface/postprocessing/template_similarity.py#L63 
Is this where you would suggest limiting to a user-set `n_channels` parameter, which would be symmetric on either side of the peak channel for that unit?